### PR TITLE
Fix #296: Add timestamps for process, identity and OTP

### LINKED ISSUE #296: Add timestamps for process, identity and OTP
We should add following timestamps to make it easier to investigate failed processes:
- `es_onboarding_process.timestamp_failed`
- `es_identity_verification.timestamp_failed`
- `es_onboarding_otp.timestamp_failed`

We should also add `es_identity_verification.timestamp_finished` to match the `es_onboarding_process` logic of saving timestamps.

### PR DESCRIPTION
**Pull request by romanstrobl**
Added timestamps for consistency and several missing error origins.

DDL changes for Postgres:
```sql
ALTER TABLE es_onboarding_process ADD COLUMN timestamp_failed TIMESTAMP;
ALTER TABLE es_onboarding_otp ADD COLUMN timestamp_failed TIMESTAMP;
ALTER TABLE es_identity_verification ADD COLUMN timestamp_finished TIMESTAMP;
ALTER TABLE es_identity_verification ADD COLUMN timestamp_failed TIMESTAMP;
```

### CODE REVIEW

File: enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/OnboardingOtpRepository.java [hunk 1]
```diff
@@ -44,7 +44,8 @@ public interface OnboardingOtpRepository extends CrudRepository<OnboardingOtpEnt
     @Query("UPDATE OnboardingOtpEntity o SET " +
             "o.status = com.wultra.app.enrollmentserver.model.enumeration.OtpStatus.FAILED, " +
             "o.timestampLastUpdated = CURRENT_TIMESTAMP, " +
-            "o.errorDetail = '" + OnboardingOtpEntity.ERROR_EXPIRED + "' " +
+            "o.errorDetail = '" + OnboardingOtpEntity.ERROR_EXPIRED + "', " +
```
Comment (saalistaja) [inline enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/OnboardingOtpRepository.java:47]: Shiouldn't we update `timestampFailed` here when the OTP expired?

**Review by saalistaja — CHANGES_REQUESTED** [review]
One redundant `setTimestampLastUpdated` found.

I am not sure if we shouldn't mark the `timestampFailed` also in the JPA queries (expiration ,...).

One more thing, food for thought, maybe all the timestamps should be inherited from the request (e.g. through `OwnerId` or some common context).

File: enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/OnboardingOtpRepository.java (hunk 1 above)
Comment (romanstrobl) [reply enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/database/OnboardingOtpRepository.java:47]: Yes, we should.

File: enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/CommonProcessLimitService.java [hunk 2]
```diff
@@ -94,6 +94,9 @@ public OnboardingProcessEntity failProcess(OnboardingProcessEntity entity, Strin
         entity.setTimestampLastUpdated(new Date());
```
[the remark this answers is not in the document]
Comment (romanstrobl) [reply enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/CommonProcessLimitService.java:94]: > I am not sure if we shouldn't mark the `timestampFailed` also in the JPA queries (expiration ,...).

Yes we should, I will go through these.

[the remark this answers is not in the document]
Comment (romanstrobl) [reply enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/CommonProcessLimitService.java:94]: > One redundant `setTimestampLastUpdated` found.
> 
> I am not sure if we shouldn't mark the `timestampFailed` also in the JPA queries (expiration ,...).
> 
> One more thing, food for thought, maybe all the timestamps should be inherited from the request (e.g. through `OwnerId` or some common context).

I am not sure this is necessary, we plan to store a complete audit record in the future, too. However, if the changes is not difficult, I will try to introduce it.

**Review by saalistaja — APPROVED** [review]
I have nothing to add to these changes. The fix looks really promising.

**Pull request merged**